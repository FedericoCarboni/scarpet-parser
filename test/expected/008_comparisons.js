/** @type {import('../../lib/Parser.js').Node} */
export const root = {
    kind: 'BinaryExpression',
    operator: ';',
    lvalue: {
        kind: 'BinaryExpression',
        operator: ';',
        lvalue: {
            kind: 'BinaryExpression',
            operator: ';',
            lvalue: {
                kind: 'BinaryExpression',
                operator: '>=',
                lvalue: {
                    kind: 'NumberLiteral',
                    literal: '1',
                    value: 1n,
                    start: {
                        offset: 24,
                        line: 1,
                        character: 0,
                    },
                    end: {
                        offset: 25,
                        line: 1,
                        character: 1,
                    },
                },
                rvalue: {
                    kind: 'NumberLiteral',
                    literal: '2',
                    value: 2n,
                    start: {
                        offset: 29,
                        line: 1,
                        character: 5,
                    },
                    end: {
                        offset: 30,
                        line: 1,
                        character: 6,
                    },
                },
                comment: '// Comparison operators\n',
                start: {
                    offset: 24,
                    line: 1,
                    character: 0,
                },
                end: {
                    offset: 30,
                    line: 1,
                    character: 6,
                },
            },
            rvalue: {
                kind: 'BinaryExpression',
                operator: '>',
                lvalue: {
                    kind: 'NumberLiteral',
                    literal: '3',
                    value: 3n,
                    start: {
                        offset: 32,
                        line: 2,
                        character: 0,
                    },
                    end: {
                        offset: 33,
                        line: 2,
                        character: 1,
                    },
                },
                rvalue: {
                    kind: 'NumberLiteral',
                    literal: '4',
                    value: 4n,
                    start: {
                        offset: 36,
                        line: 2,
                        character: 4,
                    },
                    end: {
                        offset: 37,
                        line: 2,
                        character: 5,
                    },
                },
                comment: null,
                start: {
                    offset: 32,
                    line: 2,
                    character: 0,
                },
                end: {
                    offset: 37,
                    line: 2,
                    character: 5,
                },
            },
            comment: '// Comparison operators\n',
            start: {
                offset: 24,
                line: 1,
                character: 0,
            },
            end: {
                offset: 37,
                line: 2,
                character: 5,
            },
        },
        rvalue: {
            kind: 'BinaryExpression',
            operator: '<=',
            lvalue: {
                kind: 'NumberLiteral',
                literal: '5',
                value: 5n,
                start: {
                    offset: 39,
                    line: 3,
                    character: 0,
                },
                end: {
                    offset: 40,
                    line: 3,
                    character: 1,
                },
            },
            rvalue: {
                kind: 'NumberLiteral',
                literal: '6',
                value: 6n,
                start: {
                    offset: 44,
                    line: 3,
                    character: 5,
                },
                end: {
                    offset: 45,
                    line: 3,
                    character: 6,
                },
            },
            comment: null,
            start: {
                offset: 39,
                line: 3,
                character: 0,
            },
            end: {
                offset: 45,
                line: 3,
                character: 6,
            },
        },
        comment: null,
        start: {
            offset: 24,
            line: 1,
            character: 0,
        },
        end: {
            offset: 45,
            line: 3,
            character: 6,
        },
    },
    rvalue: {
        kind: 'BinaryExpression',
        operator: '<',
        lvalue: {
            kind: 'NumberLiteral',
            literal: '7',
            value: 7n,
            start: {
                offset: 47,
                line: 4,
                character: 0,
            },
            end: {
                offset: 48,
                line: 4,
                character: 1,
            },
        },
        rvalue: {
            kind: 'NumberLiteral',
            literal: '8',
            value: 8n,
            start: {
                offset: 51,
                line: 4,
                character: 4,
            },
            end: {
                offset: 52,
                line: 4,
                character: 5,
            },
        },
        comment: null,
        start: {
            offset: 47,
            line: 4,
            character: 0,
        },
        end: {
            offset: 52,
            line: 4,
            character: 5,
        },
    },
    comment: null,
    start: {
        offset: 24,
        line: 1,
        character: 0,
    },
    end: {
        offset: 52,
        line: 4,
        character: 5,
    },
};
