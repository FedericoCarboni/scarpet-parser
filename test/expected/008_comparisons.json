{
  "errors": [],
  "warnings": [],
  "root": {
    "kind": "BinaryExpression",
    "operator": ";",
    "lvalue": {
      "kind": "BinaryExpression",
      "operator": ";",
      "lvalue": {
        "kind": "BinaryExpression",
        "operator": ";",
        "lvalue": {
          "kind": "BinaryExpression",
          "operator": ">=",
          "lvalue": {
            "kind": "NumberLiteral",
            "literal": "1",
            "value": "1",
            "start": {
              "pos": 24,
              "row": 1,
              "col": 0
            },
            "end": {
              "pos": 25,
              "row": 1,
              "col": 1
            }
          },
          "rvalue": {
            "kind": "NumberLiteral",
            "literal": "2",
            "value": "2",
            "start": {
              "pos": 29,
              "row": 1,
              "col": 5
            },
            "end": {
              "pos": 30,
              "row": 1,
              "col": 6
            }
          },
          "comment": "// Comparison operators\n",
          "start": {
            "pos": 24,
            "row": 1,
            "col": 0
          },
          "end": {
            "pos": 30,
            "row": 1,
            "col": 6
          }
        },
        "rvalue": {
          "kind": "BinaryExpression",
          "operator": ">",
          "lvalue": {
            "kind": "NumberLiteral",
            "literal": "3",
            "value": "3",
            "start": {
              "pos": 32,
              "row": 2,
              "col": 0
            },
            "end": {
              "pos": 33,
              "row": 2,
              "col": 1
            }
          },
          "rvalue": {
            "kind": "NumberLiteral",
            "literal": "4",
            "value": "4",
            "start": {
              "pos": 36,
              "row": 2,
              "col": 4
            },
            "end": {
              "pos": 37,
              "row": 2,
              "col": 5
            }
          },
          "comment": null,
          "start": {
            "pos": 32,
            "row": 2,
            "col": 0
          },
          "end": {
            "pos": 37,
            "row": 2,
            "col": 5
          }
        },
        "comment": "// Comparison operators\n",
        "start": {
          "pos": 24,
          "row": 1,
          "col": 0
        },
        "end": {
          "pos": 37,
          "row": 2,
          "col": 5
        }
      },
      "rvalue": {
        "kind": "BinaryExpression",
        "operator": "<=",
        "lvalue": {
          "kind": "NumberLiteral",
          "literal": "5",
          "value": "5",
          "start": {
            "pos": 39,
            "row": 3,
            "col": 0
          },
          "end": {
            "pos": 40,
            "row": 3,
            "col": 1
          }
        },
        "rvalue": {
          "kind": "NumberLiteral",
          "literal": "6",
          "value": "6",
          "start": {
            "pos": 44,
            "row": 3,
            "col": 5
          },
          "end": {
            "pos": 45,
            "row": 3,
            "col": 6
          }
        },
        "comment": null,
        "start": {
          "pos": 39,
          "row": 3,
          "col": 0
        },
        "end": {
          "pos": 45,
          "row": 3,
          "col": 6
        }
      },
      "comment": null,
      "start": {
        "pos": 24,
        "row": 1,
        "col": 0
      },
      "end": {
        "pos": 45,
        "row": 3,
        "col": 6
      }
    },
    "rvalue": {
      "kind": "BinaryExpression",
      "operator": "<",
      "lvalue": {
        "kind": "NumberLiteral",
        "literal": "7",
        "value": "7",
        "start": {
          "pos": 47,
          "row": 4,
          "col": 0
        },
        "end": {
          "pos": 48,
          "row": 4,
          "col": 1
        }
      },
      "rvalue": {
        "kind": "NumberLiteral",
        "literal": "8",
        "value": "8",
        "start": {
          "pos": 51,
          "row": 4,
          "col": 4
        },
        "end": {
          "pos": 52,
          "row": 4,
          "col": 5
        }
      },
      "comment": null,
      "start": {
        "pos": 47,
        "row": 4,
        "col": 0
      },
      "end": {
        "pos": 52,
        "row": 4,
        "col": 5
      }
    },
    "comment": null,
    "start": {
      "pos": 24,
      "row": 1,
      "col": 0
    },
    "end": {
      "pos": 52,
      "row": 4,
      "col": 5
    }
  }
}